src/main/java/cascading/flow/hive/HiveQueryRunner.java

```diff
@@ -65,8 +83,42 @@ public HiveQueryRunner( String queries[] )
    */
   HiveQueryRunner( HiveDriverFactory driverFactory, String queries[] )
     {
+    this( driverFactory, queries, false );
+    }
+
+  /**
+   * Constructs a new HiveQueryRunner with the given HiveDriverFactory and queries.
+   *
+   * @param driverFactory     The HiveDriverFactory to use.
+   * @param queries           The queries to run.
+   * @param fetchQueryResults {@code true} if the results of each queries must be fetched, {@code false} otherwise.
+   */
+  @SuppressWarnings("unchecked")
+  HiveQueryRunner( HiveDriverFactory driverFactory, String queries[], boolean fetchQueryResults )
+    {
     this.driverFactory = driverFactory;
     this.queries = queries;
+    this.fetchQueryResults = fetchQueryResults;
+    queryResults = fetchQueryResults ? new List[queries.length] : null;
+    }
+  
+  /**
+   * Returns an array where each position of the array is a {@link List} of rows for the query provided at the same
+   * position, i.e. the results of query in position 0 will be in position 0 of the returned array.
+   * <p>
+   * Each column value is the equivalent {@link String} of the actual value, i.e. no type is provided.
+   * </p>
+   * <p>
+   * This method can only be invoked if {@code fetchQueryResults} was set to {@code true} at construction time.
+   * </p>
+   * @return A {@link List} of rows each executed query.
+   * @throws IllegalStateException If {@code fetchQueryResults} was set to false in the constructor.
+   */
```

Comment:
I don't fully understand the comment. Could you make the javadoc a bit more clear?